bodhi/server/services/updates.py

```diff
@@ -349,17 +362,27 @@ def new_update(request):
         releases = set()
         builds = []
 
-        # Create the RpmPackage and Build entities
+        # Create the Package and Build entities
         for nvr in data['builds']:
             name, version, release = request.buildinfo[nvr]['nvr']
             # The package will have been created by validate_acls
-            package = request.db.query(RpmPackage).filter_by(name=name).one()
+            package = request.db.query(Package).filter_by(name=name).one()
+
+            # Figure out what kind of build this should be
+            try:
+                build_class = ContentType.infer_content_class(
+                    base=Build, build=request.buildinfo[nvr]['info'])
+            except Exception as e:
+                error = 'Unable to infer content_type.  %r' % e.message
+                log.exception(error)
+                request.errors.add('body', 'builds', error)
+                return
```

Comment:
I just removed the handler rather than try to mock things out.

The issue is that the error *should* always be caught in the validator before hand when inferring the Package type.